packages/flutter/lib/src/cupertino/text_selection_toolbar.dart

```diff
@@ -979,6 +1008,21 @@ class _RenderCupertinoTextSelectionToolbarItems extends RenderBox with Container
       }
     });
 
+    // If needed, re-layout slotted children using max. child height to allow
+    // the back/next buttons to be centered vertically.
+    if (greatestHeight > _backButton!.size.height) {
```

Comment:
Since the menu items are usually text, I think the children here should be baseline-aligned by default? Then this second layout pass is probably no longer needed.

Reply:
IMHO the toolbar should be child type agnostic (e.g. in iOS we already have the LiveText button which uses CustomPaint).

Reply:
In that case center aligning sounds like a better option than top aligning the children? Looks nicer when one child is significantly taller.